README.md

```diff
@@ -21,12 +20,128 @@ Or install it yourself as:
 $ gem install wor-paginate
 ```
 
+Then you can run `rails generate wor:paginate:install` to create the initializer for configuration details, including default formatter class, page and limit params and default page number.
+
+## Usage
+### Basic usage
+The basic use case is to paginate using default values. This is achieved by including the module in a controller and calling render_paginate in the method that needs pagination to be done.
+```ruby
+  class DummyModelsController < ApplicationController
+    include Wor::Paginate
+
+    def index
+        render_paginated DummyModel
+    end
+  end
+```
+The first parameter to render_paginated can be multiple things:
+* ActiveRecord/ActiveRecord::Relation
+* Enumerables (for example, arrays and ranges)
+* Pre-paginated Kaminari or will_paginate relations (original pagination will be ignored)
+
+The response to the index will then be
+```json
+{
+  "items": [
+    {
+      "id": 1,
+      "name": "1c",
+      "something": 27
+    },
+    {
+      "id": 2,
+      "name": "i",
+      "something": 68
+    },
+    ...
+    {
+      "id": 25,
+      "name": "2m",
+      "something": 32
+    }
+  ],
+  "count": 25,
+  "total_pages": 2,
+  "total_count": 28,
+  "current_page": 1,
+
+}
+```
+
+Page number is passed through the `page` option of the `render_paginated` method. If none is supplied, `params[:page]` will be used (or the default parameter configured in the initializer). Default is 1.
```

Comment:
I'd point out 1 is the default value somewhere else as someone may overlook it here.